sentry-android-integration-tests/sentry-uitest-android/src/androidTest/java/io/sentry/uitest/android/SdkInitTests.kt

```diff
@@ -74,4 +77,109 @@ class SdkInitTests : BaseUiTest() {
             assertNoOtherEnvelopes()
         }
     }
+
+    @Test
+    fun doubleInitDoesNotWait() {
+        relayIdlingResource.increment()
+        relayIdlingResource.increment()
+        // Let's make the first request timeout
+        relay.addResponse { MockResponse().setSocketPolicy(SocketPolicy.NO_RESPONSE) }
+
+        initSentry(true) { options: SentryAndroidOptions ->
+            options.tracesSampleRate = 1.0
+        }
+
+        Sentry.startTransaction("beforeRestart", "emptyTransaction").finish()
+
+        // We want the SDK to start sending the event. If we don't wait, it's possible we don't send anything before the SDK is restarted
+        Thread.sleep(500)
```

Comment:
would it make sense to have a `@TestOnly` hook inside our transport instead? 500ms sounds like a lot of waiting time at least.

Reply:
Looks like an overkill honestly 😅 
we can reduce the sleep to 100ms, it should work fine.
Let me check if i can use some hook (maybe one in the test's mock relay), if not i'll reduce the sleep

Reply:
I made the idlingResource decrement when receiving a network call and responding with a timeout, so i'll just use that, and it will work